extractors/bookseries/karelians/extractors/religion_extractor.py

```diff
@@ -0,0 +1,26 @@
+import re
+
+import core.utils.regex_utils as regexUtils
+from core.pipeline_construction.base_extractor import BaseExtractor
+
+
+class ReligionExtractor(BaseExtractor):
+    extraction_key = 'religion'
+
+    def __init__(self, cursor_location_depends_on=None, options=None):
+        super(ReligionExtractor, self).__init__(cursor_location_depends_on, options)
+        self.RELIGION_PATTERN = r"(?P<ortodoksi>ortodoksi)"

```

Comment:
I think this could be named as `OrthodoxExtractor` or something like that. and the resulting key be `isOrthodox` or `orthodox` depending the existing convention with boolean flags.

My reasoning with using orthodox in name is that most likely everyone in the data are lutherian christians unless otherwise stated so probably we do not need a _religion extractor_ as such...? You can ask from researchers if this assumption is okay for them.

Reply:
I did a quick search to the text files with ortodoksi keyword and got varying results about people being part of some orthodoxy church activities. For example "XXX on ortodoksisen kirkon isännöitsijä" or someone being in the choir of a church. While I guess it is safe to assume them also belonging to said church, I wonder if the more accurate name for field of this data would be something like `affiliatedToOrthodoxChurch` etc? That would hopefully tell researchers more clearly that the person in question is somehow involved with the church activities and not necessarily just in the faith.

Reply:
My initial thinking was to just drop this religion extractor, as it was just a practice/example extractor, but I like this `affiliatedToOrthodoxChurch` thinking. Maybe some researcher will get something out of just knowing that there is some link between the person and the Orthodox church. Will have to think about this a bit more :thinking: There's still the question of whether this can be implemented reliably enough without utilizing NLP, and that is still coming :sweat_smile: